### PR TITLE
Unify upload pre- and post-processing logic

### PR DESCRIPTION
**Pull request by subdavis**
* Moves ownership of all the pre- and post-processsing of upload groups to girder.
* Removes #196 (sorry Matt) because the server is a better place to decide whether or not a group of potential file uploads is vaid.
* Automatically parse CSV and YML annotations into our internal JSON immediately and store as an item rather than computing this on-demand.  This means that we now have a nice method of performing "bulk import" that can be done entirely server side.
* This will make it easier to extract the KPF parsing into a girder worker task (since it takes several seconds of CPU-intensive work) but I haven't done that yet. 

Still WIP testing, but comments are welcome.  

Removes ~160 LoC.

### CODE REVIEW

**Comment by subdavis** [comment]
@mattrix27 let me know what you think.  

It looks like there's still an issue with FPS: for videos it's set to 5 by the transcoding pipeline, but it should actually be calculated from the KPF files.  This is a problem for another day: setting "30" as the framerate through girder UI fixes the problem.

File: client/src/views/Home.vue [hunk 1]
```diff
@@ -119,53 +106,6 @@ export default {
         this.uploaderDialog = true;
       }
     },
-    uploaded(uploads) {
-      this.uploaderDialog = false;
-      // Check if any transcoding should be done
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/src/views/Home.vue:124]: I also hulk-smashed the friendly snackbar notice that a job had been started.  I'll bring back the event and close the uploader.

File: server/viame_server/viame.py [hunk 2]
```diff
@@ -82,48 +75,107 @@ def run_pipeline_task(self, folder, pipeline):
...
+        csvItems = Folder().childItems(folder, filters={"lowerName": {"$regex": csvRegex}})
+        if csvItems.count() == 1:
+            file = Item().childFiles(csvItems.next())[0]
+            saveTracks(folder, viame_serializer.load_csv_as_tracks(file), user)
+        elif csvItems.count() > 1:
+            raise RestException('Can have at most 1 annotation CSV file')
+
+        ymlItems = Folder().childItems(folder, filters={"lowerName": {"$regex": ymlRegex}})
+
+        # Proprocess KPF if necessary
```
Comment (mattrix27) [inline server/viame_server/viame.py:171]: Minor typo?

**Comment by mattrix27** [comment]
The changes make sense to me. It fixes a lot of things that Jacob mentioned about the pre-processing the kpf conversion. Mainly for me, but is it okay that the kpf conversion is not a convert task like convert image and video?

**Comment by subdavis** [comment]
good question.   I'd say let's get this merged, create an issue, and that can be (one of) your next tasks?  Would that be OK, or are you busy with other stuff?

**Comment by mattrix27** [comment]
That would be great!

**Comment by subdavis** [comment]
Logged #246

**Comment by AlmightyYakob** [comment]
A few style suggestions, and one actual question, but besides that it looks good

File: server/viame_server/utils.py [hunk 3]
```diff
@@ -10,6 +17,12 @@
 validImageFormats = {*webValidImageFormats, "tif", "tiff", "sgi", "bmp", "pgm"}
 validVideoFormats = {"mp4", "avi", "mov", "mpg"}
 
+videoRegex = re.compile("\." + "|\.".join(validVideoFormats) + '$', re.IGNORECASE)
+imageRegex = re.compile("\." + "|\.".join(validImageFormats) + '$', re.IGNORECASE)
+safeImageRegex = re.compile("\." + "|\.".join(webValidImageFormats) + '$', re.IGNORECASE)
```
[the remark this answers is not in the document]
Comment (subdavis) [reply server/viame_server/utils.py:22]: I tried to use f-strings, but putting a list comprehension inside an f-string looked gross.  How would you recommend to write it?

File: server/viame_server/viame.py [hunk 4]
```diff
@@ -82,48 +75,107 @@ def run_pipeline_task(self, folder, pipeline):
...
-        )
+    def validate_files(self, files):
+        ok = True
+        message = ""
+        mediatype = ""
+        videos = [f for f in files if videoRegex.search(f)]
+        csvs = [f for f in files if csvRegex.search(f)]
+        images = [f for f in files if imageRegex.search(f)]
+        ymls = [f for f in files if ymlRegex.search(f)]
+        print(videoRegex.match('foo.mp4'), videoRegex)
```
[the remark this answers is not in the document]
Comment (subdavis) [reply server/viame_server/viame.py:89]: No I like it there /s

File: server/viame_server/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (BryonLewis) [reply server/viame_server/utils.py:22]: see above my change request, I believe this needs to change besides the formatting.

File: server/viame_server/utils.py [hunk 5]
```diff
@@ -10,6 +17,12 @@
...
+csvRegex = re.compile("\.csv$", re.IGNORECASE)
```
[the remark this answers is not in the document]
Comment (subdavis) [reply server/viame_server/utils.py:23]: `videoRegex = re.compile("(\." + "|\.".join(validVideoFormats) + ')$', re.IGNORECASE)`  ?

**Pull request merged**